automations/python/label_pr.py

```diff
@@ -76,21 +80,54 @@ def get_pull_request(gh: Github, html_url: str) -> PullRequest:
     return gh.get_organization(org).get_repo(repo).get_pull(pr_num)
 
 
+def get_authenticated_html(url: str) -> str:
+    """
+    Login to the GitHub UI using the username and password, followed by 2FA. Then
+    navigate to the specified URL as the authenticated user and scrape the text body.
+
+    :param url: the URL to scrape after authenticating in GitHub
+    :return: the text content of the scraped page
+    """
+
+    browser = mechanize.Browser()
+    browser.set_cookiejar(CookieJar())
+    browser.open("https://github.com/login/")
+    browser.select_form(nr=0)
+    browser.form["login"] = os.getenv("GH_LOGIN")
+    browser.form["password"] = os.getenv("GH_PASSWORD")
+    browser.submit()
+
+    browser.select_form(nr=0)
```

Comment:
Just for posterity, it would be great to have a description of this and the step at L95, since `nr=0` doesn't provide a whole lot of context

Reply:
Added the comment in 6253ac3.